CMakeLists.txt

```diff
@@ -59,270 +58,254 @@ include (GitRepo)
 set(xournalpp_CXX_FLAGS ${CMAKE_CXX_FLAGS})
 
 if (MSVC)
-	set(xournalpp_CXX_FLAGS ${xournalpp_CXX_FLAGS} -DNOMINMAX -DWIN32_LEAN_AND_MEAN)
-else ()
-    set(LIB_ENDING "so")
+    set(xournalpp_CXX_FLAGS ${xournalpp_CXX_FLAGS} -DNOMINMAX -DWIN32_LEAN_AND_MEAN)
 endif ()
 
 ## Libraries ##
-
-macro(add_includes_ldflags LDFLAGS INCLUDES)
-	set(xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "${LDFLAGS}")
-    include_directories(SYSTEM ${INCLUDES})
-    # Do not add system headers and external libraries to the additional include paths
-    # set(xournalpp_INCLUDE_DIRS ${xournalpp_INCLUDE_DIRS} "${INCLUDES}")
-endmacro (add_includes_ldflags LDFLAGS INCLUDES)
-
+##############################
+# c++17 support
+##############################
 find_package(CXX17 REQUIRED COMPONENTS optional)
 find_package(Filesystem REQUIRED COMPONENTS ghc Final Experimental)
 
-find_package(X11)
-if(X11_FOUND AND X11_Xi_FOUND)
-	add_definitions (-DX11_ENABLED)
-	add_includes_ldflags ("${X11_X11_LIB}" "${X11_INCLUDE_DIR}")
-	add_includes_ldflags ("${X11_Xi_LIB}" "${X11_Xi_INCLUDE_PATH}")
-endif()
-
-# libexec
 if (WIN32)
-	set(xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "-mwindows")
-	# Need ImageMagick to generate icons on Windows
+	# libexec
 	find_package(ImageMagick REQUIRED COMPONENTS convert)
-else ()
-	find_package(Backtrace REQUIRED)
-	add_includes_ldflags("${Backtrace_LIBRARIES}" "${Backtrace_INCLUDE_DIRS}")
+else (WIN32)
+	##############################
+	# stacktrace library
+	# Todo use platform independent stacktrace library
+	# Currently this works only for linux systems
+	##############################
+    find_package(Backtrace REQUIRED)
```

Comment:
You should not load this as required within some if clause if we do not actually require it after all...